src/test/ui/parser/lex-bad-char-literals-3.stderr

```diff
@@ -8,5 +8,15 @@ help: if you meant to write a `str` literal, use double quotes
 LL |     "●●" //~ ERROR: character literal may only contain one codepoint
    |     ^^^^
 
-error: aborting due to previous error
+error[E0308]: mismatched types
+  --> $DIR/lex-bad-char-literals-3.rs:3:5
+   |
+LL |     '●●' //~ ERROR: character literal may only contain one codepoint
+   |     ^^^^ expected char, found reference
+   |
+   = note: expected type `char`
+              found type `&'static str`
```

Comment:
We should give the recovered value `TyErr` (or even keep the `char` type with a placeholder value) instead of `TyStr` to avoid these.